js/providers/Router.js

```diff
@@ -49,7 +48,7 @@ define(
         }, {});
         const routesWithRefreshedToken = Object.keys(routes).reduce((accumulator, key) => {
 					accumulator[key] = (...args) => {
-            this.setCurrentView('loading');
+            this.getModel().loading(true);
```

Comment:
I can't quite tell fro code, but is this setting a global 'loading' observable that will set a top-level 'loading' message?  I seem to recall that when the 'top' level loading is presented, the currentView component is removed from the dom. By removing it from the dom, it destroyes and re-creates the component needlessly.  If this is true, I think each component should be responsible for manging their own 'lodading' messaging and this shouldn't be managed at the route-processing level (we shoudln't see a flash of 'loading' show up when we go too cc/analysis/detail or cc/analysis/executions.

Reply:
@chrisknoll, this is setting loading icon in navbar instead of top-level loading panel to remove the re-initiation of page-component:
![image](https://user-images.githubusercontent.com/10010071/45626700-24afe700-ba5e-11e8-9a84-7bc6866de522.png)

And yes, components should take care of loading panel on their own.